libcaf_core/caf/flow/step/element_at.hpp

```diff
@@ -0,0 +1,54 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#pragma once
+
+#include "caf/fwd.hpp"
+
+#include <cstddef>
+
+namespace caf::flow::step {
+
+template <class T>
+class element_at {
+public:
+  using input_type = T;
+
+  using output_type = T;
+
+  explicit element_at(size_t ind) : element_index_(ind) {
+    // nop
+  }
+
+  element_at(element_at&&) = default;
+  element_at(const element_at&) = default;
+  element_at& operator=(element_at&&) = default;
+  element_at& operator=(const element_at&) = default;
+
+  template <class Next, class... Steps>
+  bool on_next(const input_type& item, Next& next, Steps&... steps) {
+    if (element_index_ == current_index_) {
+      if (!next.on_next(item, steps...))
```

Comment:
After finding the correct item, this operator should call `on_complete` and return `false`.